cmd/govim/vimstate.go

```diff
@@ -196,9 +196,51 @@ func (v *vimstate) popupSelection(args ...json.RawMessage) (interface{}, error)
 		return nil, nil
 	}
 
-	edit := edits[selection-1]
+	fix := fixes[selection-1]
 
-	return nil, v.applyMultiBufTextedits(nil, edit.DocumentChanges)
+	// Edits should be applied before any Command according to LSP 3.16.
+	if len(fix.edit.DocumentChanges) > 0 {
+		if err := v.applyMultiBufTextedits(nil, fix.edit.DocumentChanges); err != nil {
+			return nil, err
+		}
+	}
+
+	if fix.command != nil {
```

Comment:
We should look to DRY up this pattern now that it's repeated here and in fillstruct. Or are you thinking we do that in a later PR?

Reply:
I intentionally left it this way since I'm not sure how things will work when Rob's work with the "command API" is done.

Hopefully we can cut the entire code action turnaround in the fillstruct case. So, "yes, in a later PR" :)